pkg/actuators/client/client.go

```diff
@@ -428,3 +454,55 @@ func (c *ibmCloudClient) GetSecurityGroupsByName(securityGroupNames []string, re
 	return nil, fmt.Errorf("could not retrieve security group ids of names: %v", securityGroupMap)
 
 }
+
+// GetDedicatedHostByName retrieves Dedicated Hosts info
+func (c *ibmCloudClient) GetDedicatedHostByName(dedicatedHostName string, resourceGroupID string, zoneName string) (string, error) {
+	// Initialize List Dedicated Hosts Options
+	dedicatedHostOptions := c.vpcService.NewListDedicatedHostsOptions()
+
+	// Set Resource Group ID
+	dedicatedHostOptions.SetResourceGroupID(resourceGroupID)
+
+	// Set Zone
+	dedicatedHostOptions.SetZoneName(zoneName)
+
+	// Get a list of all Dedicated Hosts
+	dedicatedHosts, _, err := c.vpcService.ListDedicatedHosts(dedicatedHostOptions)
+	if err != nil {
+		return "", err
+	}
+
+	if dedicatedHosts != nil {
+		for _, eachDedicatedHost := range dedicatedHosts.DedicatedHosts {
+			if *eachDedicatedHost.Name == dedicatedHostName {
+				// return Dedicated Host ID
+				return *eachDedicatedHost.ID, nil
```

Comment:
We are dereferencing here without checking if the object is non-nil, I this likely to cause panics?

Reply:
Added a condition to make sure `dedicatedHosts` is not empty. From there we can assume that Name and ID are present